eZ/Publish/Core/Repository/Repository.php

```diff
@@ -879,6 +744,28 @@ public function getFieldTypeService()
     }
 
     /**
+     * Get PermissionService.
+     *
+     * @return \eZ\Publish\API\Repository\PermissionService
+     */
+    public function getPermissionService()
+    {
+        if ($this->permissionService === null) {
+            $this->permissionService = new PermissionService(
+                $this,
```

Comment:
We need it only for `sudo()`, should be possible to get rid of that once we restructure repo implementation (introduce domain logic implementation).